ui/src/app/pages/profile/profile-page.tsx

```diff
@@ -128,6 +199,39 @@ export const ProfilePage = withUserProfile()(class extends React.Component<
     }
   }
 
+  getRoleOptions(): Array<{label: string, value: InstitutionalRole}> {
+    const {institutions, currentProfile} = this.state;
+    if (currentProfile) {
+      const selectedOrgType = institutions.find(
+        inst => inst.shortName === currentProfile.verifiedInstitutionalAffiliation.institutionShortName);
+      if (selectedOrgType) {
+        const sel = selectedOrgType.organizationTypeEnum;
+
+        const availableRoles: Array<InstitutionalRole> =
+           AccountCreationOptions.institutionalRolesByOrganizationType
+               .find(obj => obj.type === sel)
+               .roles;
+
+        return AccountCreationOptions.institutionalRoleOptions.filter(option =>
+           availableRoles.includes(option.value)
+       );
+      }
+    }
+  }
+
+  updateVerifiedInstitutionRole(newRole) {
```

Comment:
This sounds like it does persistence - maybe a better name would be 'setOtherInstitutionalRoleState` or something along those lines?

Reply:
The main purpose of this method is to set or update the institution Role. If the institution Role is other , show the other text box else hide it. So renaming it to setOtherInstitutionRoleState will give an impression it's just setting Other .
 I understand your concern so i can do the following: 

- Change the method name to setInstitutionRole

- Remove the state variable showOther

- add a check institutionRole === Role.Other at l370 rather than using state variable.

Reply:
Ah, sure, I see. Misunderstood what line 224 was doing, I guess.